### ISSUE TITLE
New rule: check that a class name is not alreadu used by the .NET base classes

### ISSUE THREAD

**Issue by jairbubbles** [meziantou/Meziantou.Analyzer#253]
Ex: 

```csharp
namespace Test
{
  public class Action // <-- Warning, this will clash with System.Action
  {
  }
}
```

I don't know if all classes should be taken in account, it could be limited to `System`  namespace first.

**Comment by meziantou** [comment]
MA0104 should detect this case. Note that you need to manually enable the rule.

https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0104.md

**Comment by jairbubbles** [comment]
Wow, you're fast! Thx
